packages/manager/apps/pci-public-ip/src/pages/order/hooks/useFloatingRegions.ts

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { TRegion } from '@/pages/order/types';
+import { useAvailabilities } from '@/pages/order/hooks/useAvailabilities';
+
+const getMacroRegion = (region: string) => {
+  const localZonePattern = /^lz/i;
+  let macro;
+  if (
+    localZonePattern.test(
+      region
+        .split('-')
+        ?.slice(2)
+        ?.join('-'),
+    )
+  ) {
+    // The pattern for local zone is <geo_location>-LZ-<datacenter>-<letter>
+    // geo_location is EU-WEST, EU-SOUTH, maybe ASIA-WEST in the future
+    // datacenter: MAD, BRU
+    macro = /[\D]{2,3}/.exec(
+      region
+        .split('-')
+        ?.slice(3)
+        ?.join('-'),
+    );
+  } else {
+    macro = /[\D]{2,3}/.exec(region);
+  }
+  return macro ? macro[0].replace('-', '').toUpperCase() : '';
+};
+
+export const useFloatingRegions = (projectId: string) => {
+  const { t: tRegion } = useTranslation('regions');
+  const { data: availabilities, isPending } = useAvailabilities(projectId);
+
+  const [regions, setRegions] = useState([]);
+
+  useEffect(() => {
+    if (availabilities) {
+      const plansRegions = availabilities.plans
+        .find(({ code }) => code === 'floatingip.floatingip.hour.consumption')
+        ?.regions.map((region: TRegion) => {
+          const plansRegion = { ...region };
+
+          const { name } = plansRegion;
+          const macro = tRegion(
+            `manager_components_region_${getMacroRegion(name)}`,
+          );
+          const micro = tRegion(
+            `manager_components_region_${getMacroRegion(name)}_micro`,
+            { micro: name },
+          );
+          const continent = tRegion(
+            `manager_components_region_continent_${getMacroRegion(name)}`,
+          );
+
+          plansRegion.microName = micro;
```

Comment:
can you be more explicit?does it change something if it's a const?if there's some global recommendation about this subject, an eslint rule should be added instead, and if i use let in this context the linter asks me to change it back to const, cause the reference doesn't change.
Change requests never bother me, one should never stop learning, i comment only when the change requests are worthy and make sens, and i recall here your comment about using react interpolation, sometimes it just looks it's not a collaboration anymore, CRs are meant to help making the product better, not the other way around

Reply:
Man, that doesn't even tackle your first point of let, you're dragging me into a childish game here, just stop it already.

`
return {
      ...region,
      macroName: tRegion(
        `manager_components_region_${getMacroRegion(region.name)}`,
      ),
      microName: tRegion(
        `manager_components_region_${getMacroRegion(region.name)}_micro`,
        { micro: region.name },
      ),
      continent: tRegion(
        `manager_components_region_continent_${getMacroRegion(
          region.name,
        )}`,
      ),
    }
`
Should i start doing the same thing to your PR's?especially where your refactor a component by moving its jsx into another one and giving it 13 props while it's not reused instead of just moving the logic to a separate hook?
It ain't a competition, every random guy can come up with better suggestion than yours and mine anyway, we are collaborating and moving together to a same common goal:making the manager better, so each time you do something ask yourself: is my action creating a value for the product?is it making it better?or am i just trying to show them all who is the boss?
It's been a while since you're throwing those reviews to the team, we were simply ignoring them when they don't make sense, we are not supposed to justify why we reject a dumb suggestion like that knwoing that we receive so many per each PR.
your reviews are always welcome when they make sens, i'm talking on behalf of the whole migration team, sorry for being dragged to this low level but you insisted on making it personal,polluting an opensource project's PR like that, and since no one could ask you to stop, since you seem to make part of the "master team" that owns the codebase, i had to take action.
Please grow up